CardsOfOpprotunity/CardsOfOpprotunity/Views/BoardGame/BoardGameView.swift

```diff
@@ -9,51 +9,160 @@ import SwiftUI
+                            .padding()
+                        Text("Score: \(viewModel.playerOneScore)")
+                            .font(.title2)
+                            .fontWeight(.bold)
+                            .padding()
+                    }
+                    
+                    LoadingButton(title: viewModel.buttonTitle) {
+                        do {
+                            try await viewModel.revealCards()
+                            //showingResults = true
+                            viewModel.isShowingHand = true
+                        } catch {
+                            
+                        }
+                    }
+                    .disabled(viewModel.userSelectedCards)
+                    .opacity(viewModel.userSelectedCards ? 0.8 : 1)
+                    .animation(.easeInOut, value: viewModel.userSelectedCards)
+                    .padding()
+                    
+                    HStack {
+                        Text("Player 2: \(gameData.playerTwoName)")
+                            .frame(maxWidth: .infinity, alignment: .leading)
+                            .padding(.leading)
+                            .font(.title2.bold())
+                        Text("Score: \(viewModel.playerTwoScore)")
+                            .font(.title2)
+                            .fontWeight(.bold)
+                            .padding()
+                    }
+                    
+                    CardSelectionView(selectedCardIndex: $viewModel.secondSelectedCardIndex, isOffsetPositive: false)
+                        .scaledToFit()
                 }
-                .disabled(viewModel.userSelectedCards)
-                .opacity(viewModel.userSelectedCards ? 0.6 : 1)
-                .animation(.easeInOut, value: viewModel.userSelectedCards)
                 Spacer()
-            }
+                }
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+            .navigationBarBackButtonHidden(true)
+            .alert(isPresented: $viewModel.showAlert) {
+                Alert(title: Text("Error"), message: Text(viewModel.alertMessage), dismissButton: .default(Text("OK")))
         }
-        .alert(isPresented: $viewModel.showAlert) {
-            Alert(title: Text("Error"), message: Text(viewModel.alertMessage), dismissButton: .default(Text("OK")))
+            if viewModel.isShowingHand {
+                ResultsView(viewModel: viewModel)
+                    .frame(width: 350, height: 700)
+                    .background(Color.white)
+                   // .border(Color.black, width: 2)
+                    .cornerRadius(20).shadow(radius:20)
+                    .overlay(
+                        RoundedRectangle(cornerRadius: 20)
+                            .stroke(Color.black, lineWidth: 5)
+                    )
+            }
         }
+        .transition(.opacity)
+        .animation(.easeInOut, value: viewModel.isShowingHand) // Animate the transition
```

Comment:
Comments are totally cool, but they should be left in the project if the line of code needs explaining. " // Animate the transition" doesn't really need to be part of the code base.